Search all fields in full text search

johntseng opened the pull request:
There are two commits here that fix #184.

One creates a full text index over displayXhtml so that all fields are searchable. The other fills in mainHeadWord from headword as necessary.

This should make it so that we have backwards compatibility with 9.0 because the existing version of Webonary searches all fields.

This also lays the foundation for future field specific searches by filling in missing data from the data supplied.

johntseng committed:
Change full text search to search displayXhtml instead of only main headword and definitions.

johntseng committed:
Refactor mongo test setup into a separate file so it can be reused in other tests.

johntseng committed:
Use headword if mainHeadWord is empty.

phillip-hopper reviewed (changes requested):
I am not able to build. This is the output:
![image](https://user-images.githubusercontent.com/5262964/173682284-9fcdea89-d576-49a6-98a1-709031c0fe88.png)

johntseng committed:
Remove failing test.

This test succeeds the first time it is run because a snapshot doens't exist, yet. All subsequent runs fail because the snapshot is different on every single run.

johntseng committed:
Remove isConnected check.

MongoClient.isConnected has been deprecated. It will now (re)connect automatically. https://mongodb.github.io/node-mongodb-native/3.6/reference/unified-topology/#mongoclient-connect-isconnected

johntseng committed:
Remose useNewUrlParser.

MongoClientOptions.useNewUrlParser has been removed: https://github.com/mongodb/node-mongodb-native/pull/2623

johntseng committed:
Update all packages and update code to compile.

Updated all the packages with npm-check-updates, then fixed all the compile errors.

johntseng committed:
Remove unused import.

johntseng committed:
Remove unnecessary jest configuration.

This is causing an error when running tests with the updated version of jest.

johntseng committed:
Rollback AWS CDK v2 packages to v1.

Most of our stuff uses v1, but two packages were updated to v2, causing incompatibilities.

Updating to v2 requires some pretty significant changes, so I'll table that for now.

johntseng committed:
Merge remote-tracking branch 'sillsdev/develop' into issue-184

# Conflicts:
#	webonary-cloud-api/package-lock.json

johntseng committed:
Reinstall packages to update package-lock.json.
